### PR TITLE
storage/remote: Spawn not more than "maxConcurrentSends" goroutines.

### PR DESCRIPTION
**Pull request by pydima**
Fixes https://github.com/prometheus/prometheus/issues/1622

@beorn7 @fabxc @juliusv

### CODE REVIEW

File: storage/remote/queue_manager_test.go [hunk 1]
```diff
@@ -82,3 +109,40 @@ func TestSampleDelivery(t *testing.T) {
...
+	m := NewStorageQueueManager(c, n)
+
+	go m.Run()
+
+	for _, s := range samples {
+		m.Append(s)
+	}
+
+	for i := 0; i < maxConcurrentSends; i++ {
+		c.getData <- true // wait while all goroutines are spawned
```
Comment (brian-brazil) [inline storage/remote/queue_manager_test.go]: Comments should be full sentences. Start with a capital letter and end with a full stop.

**Comment by brian-brazil** [comment]
:+1:

**Comment by pydima** [comment]
@brian-brazil Comment message is fixed.

**Comment by brian-brazil** [comment]
Thanks!

**Comment by guanglinlv** [comment]
:+1: :+1: :+1:

**Pull request merged**